github-management/default-branch-migration.md

```diff
@@ -0,0 +1,127 @@
+---
+title: Default Branch Migration
+linkTitle: Default Branch Migration
+description:  |
+  Instructions on how to migrate the default branch from master to main.
+weight: 99
+type: docs
+aliases: [ "/rename" ]
+---
+
+# Default Branch Migration
+
+This document outlines steps needed to migrate the default branch
+of your repo from `master` to `main`.
+
+Note: This document is currently a work in progress.
+
+If you have questions about the process, reach out to the GitHub Management Team
+on the [#github-management] channel on slack or open an issue in the [kubernetes/org] repo.
+
+## Prerequisites
+
+- [ ] Ensure that your repo doesn't have too many:
+  - [ ] open PRs. The branch rename will re-trigger prow on _all_ PRs,
+    which will cause a huge spike in the CI load.
+  - [ ] periodic jobs. Changes to frequent periodics haven't been tested yet.
+
+- [ ] Create an issue in the [kubernetes/org] repo to track the branch
+rename for your repo.
+
+- [ ] Assign the GitHub Management team ([@kubernetes/owners]) for approval.
+
+- [ ] Once the issue has been approved, send a notice to your SIG's
+mailing list about the potential branch rename.
+
+## Changes pre-rename
```

Comment:
Here's the only thing I would block on. Split this into non-disruptive and disruptive, or anytime and just before branch rename. As much as possible, the goal should be to minimize the number of changes that need to happen in lockstep with branch rename. Makes it easier to allow more people to help out up front for a smoother experience.

Reply:
And tbh I think GitHub admin approval would be required once all the non disruptive stuff is done, or the list of disruptive changes has been identified, as in "ok you are ready to go ahead" though a "your plan looks sound" before implementation is ok too (I'm just trying to minimize blocking)

Reply:
> Split this into non-disruptive and disruptive, or anytime and just before branch rename.

Done.

> And tbh I think GitHub admin approval would be required once all the non disruptive stuff is done, or the list of disruptive changes has been identified

Done.